### PR TITLE
Add generatedTime property in json output

### PR DESCRIPTION
**Pull request by mixed**
I need generated time in json format when I handle data for analysis. Html format supported [generated time](https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/report/report-generator.js#L44-L52). I need same thing in json format.

What do you think about?
Thank you.

### CODE REVIEW

**Comment by paulirish** [comment]
Nice! This looks great.

image

I think the only question is if we'd want this timestamp in the JSON that gets passed off into other modules that consume Lighthouse. (e.g. `lighthouseVersion` is set in the results object back in `runner.js`)
But IMO the report timestamp feels local to the report.. so I think we're good here. :)

**Comment by ebidel** [comment]
Love it.

+1 to generating this timestamp and adding it in runner.js. I think everywhere `lighthouseVersion` appears, this should also be present :)

@mixed check out https://github.com/GoogleChrome/lighthouse/pull/749/files for all the places I added `lighthouseVersion`.

**Comment by addyosmani** [comment]
> +1 to generating this timestamp and adding it in runner.js. I think everywhere lighthouseVersion appears, this should also be present :)

👍

**Comment by brendankenny** [comment]
lol, mixed signals. Sorry @mixed 

I think I agree with Eric here that the generated time should just be put on the output from `runner.js`. Anyone consuming Lighthouse as a module and who don't want it can just ignore.

(incidentally, we should probably rename `report-generator.js` to `html-report-generator.js` at some point since it generates only the html right now, and that's kind of nice)

File: lighthouse-core/report/report-generator.js [hunk 1]
```diff
@@ -107,6 +99,20 @@ class ReportGenerator {
   }
 
   /**
+   * Generated time of report
+   * @return {string}
+   */
+  _generatedTime() {
+    const options = {
```
Comment (samccone) [inline lighthouse-core/report/report-generator.js]: recreating these static options and formatter instance seems unneeded. Consider pulling out into module global and private vars

Comment (paulirish) [inline lighthouse-core/report/report-generator.js]: It's called just twice (typically); not a big deal. 
Let's leave this alone for now, especially this PR doesn't change what was already there. :)

**Comment by mixed** [comment]
@paulirish 
OK. I will update your guys review. Thank you for reviews.

**Comment by mixed** [comment]
@paulirish 
I found a problem. `html locale` is different from `json locale` in that I live in korea. What do you think is the problem?

**Comment by paulirish** [comment]
> I found a problem. html locale is different from json locale in that I live in korea. What do you think is the problem?

@mixed I don't totally understand where the problem is so far... :)

This is my current understanding of things:

* `toJSON()` creates a timestamp via toISOString(), where the timezone is in UTC + 0.
* later in `_formatTime`, DateTimeFormat formats that time and returns something like `"10/6/2016, 2:17:01 PM PDT"`
  * with a timezone of korea, i think _formatTime should return something like `"10/7/2016, 6:16:29 AM GMT+9"`? (I just added ` timeZone: 'Asia/Seoul'` to the `options` object)
* Lastly, we hardcode `"en-US"` in [`Intl.DateTimeFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DateTimeFormat) but we could leave it out, and it'd select the local locale. So for the implicit `"ko"`, you'd get `"2016. 10. 6. 오후 2시 20분 28초 GMT-7"`. I'm +1 on making that change.

Am I missing anything else?

**Comment by paulirish** [comment]
Ah! I understand the problem!! The failing test. :)  

Yes the locale of Travis is different than yours. Okay we can fix that easily. :)

**Comment by paulirish** [comment]
Fixed the locale issue in https://github.com/GoogleChrome/lighthouse/pull/752/commits/855dc3958c5fdba7896cc036b9f2651eee28385a ;)

Let's land this!

**Comment by mixed** [comment]
@paulirish 
Thanks @paulirish

**Comment by samccone** [comment]
@mixed thanks for the PR :clap:

**Pull request merged**